Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -401,7 +408,16 @@ static inline void GeneratePixelShader(T& out, DSTALPHA_MODE dstAlphaMode, API_T
 
 		// On GLSL, input variables must not be assigned to.
 		// This is why we declare these variables locally instead.
-		out.Write("\tfloat4 col0, col1;\n");
+		if (ApiType == API_OPENGL)
+		{
+			out.Write("\tfloat4 col0, col1;\n");
+		}
+		// On D3D, input variables must be completely initialized before being used.
+		else
+		{
+			out.Write("\tfloat4 col0 = 0.0;\n");
+			out.Write("\tfloat4 col1 = 0.0;\n");
+		}
```

Comment:
Why do you need globals? Isn't it fine to just declare them within the main function?